Bio/Phenotype/PhenMicro.py

```diff
@@ -0,0 +1,1101 @@
+    (61.853516785566917, 'logistic')
+    
+    If not sigmoid function is specified, the first one that is succesfully
+    fitted is used. The user can also specify a specific function.
+    
+    >>> well.fit('gompertz')
+    >>> print(well.slope, well.model)
+    (127.94630059171354, 'gompertz')
+    
+    If no function can be fitted, the parameters are left as None, except for
+    the max, min, average_height and area. 
+    """
+    
+    def __init__(self, wellid, plate=PlateRecord(None), signals={}):
+        self.plate = plate
+        
+        self.id = wellid
+    
+        # Curve parameters (to be calculated with the "fit" function)
+        # Parameters that don't need scipy
+        self.max = None
+        self.min = None
+        self.average_height = None
+        
+        # Parameters that need scipy
+        self.area = None
+        self.plateau = None
+        self.slope = None
+        self.lag = None
+        self.v = None
+        self.y0 = None
+        self.model = None
+    
+        # Original signals (private)
+        self._signals = signals
+    
+    def _interpolate(self, time):
+        """Private method to get a linear interpolation of the signals
+        at certain time points.
+        """
+        times = sorted(self._signals.keys())
+        
+        return np.interp(time,
+                        times,
+                        [self._signals[x] for x in times],
+                        left=np.nan, right=np.nan)
+    
+    def __setitem__(self, time, signal):
+        """Assign a signal at a certain time point.
+        """
+        try:
+            time = float(time)
+        except ValueError:
+            raise ValueError('Time point should be a number')
+        try:
+            signal = float(signal)
+        except ValueError:
+            raise ValueError('Signal should be a number')
+    
+        self._signals[time] = signal
+    
+    def __getitem__(self, time):
+        """Returns a subset of signals or a single signal.
+        """
+        if isinstance( time, slice ):
+            # Fix the missing values in the slice
+            if time.start is None:
+                start = 0
+            else:
+                start = time.start
+                
+            if time.stop is None:
+                stop = max(self.get_times())
+            else:
+                stop = time.stop
+            
+            time = np.arange(start, stop, time.step)
+            return list(self._interpolate(time))
+            
+        elif isinstance( time, int ) or isinstance( time, float ): 
+            return self._interpolate(time)
+            
+        raise ValueError('Invalid index')
+        
+    def __iter__(self):
+        for time in sorted(self._signals.keys()):
+            yield time, self._signals[time]
+    
+    def __eq__(self, other):
+        if isinstance(other, self.__class__):
+            if self._signals.keys() != other._signals.keys():
```

Comment:
This looks like it would only work with Python 2 dictionaries, under Python 3 you would get iterators which would not compare as equal even if the list of keys was. This tells me we need more tests.

Probably you need  ``if list(self._signals) != list(other._signals):`` (since iterating over a dict yields the keys)